components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/UsageClient.java

```diff
@@ -0,0 +1,217 @@
+    private static APIUsageStatisticsClient usageStatisticsClient;
+
+    /**
+     * central point to initialise datasources or related configuration done by the admin-dashboard analytics section
+     *
+     * @throws APIMgtUsageQueryServiceClientException
+     */
+    public static void initializeDataSource() throws APIMgtUsageQueryServiceClientException {
+
+        try {
+            APIUsageStatisticsClient client = UsageClient.getStatisticClient("");
+            client.initializeDataSource();
+        } catch (APIMgtUsageQueryServiceClientException e) {
+            throw new APIMgtUsageQueryServiceClientException("Error in initializing data sources", e);
+        }
+
+    }
+
+    /**
+     * central public method used to get the instance if the statistic client
+     *
+     * @return return the APIUsageStatisticsClient implementation
+     * @throws APIMgtUsageQueryServiceClientException if error in creating instance
+     */
+    public static APIUsageStatisticsClient getClient(String user) throws APIMgtUsageQueryServiceClientException {
+        if (isDataPublishingEnabled()) {
+            try {
+                return UsageClient.getStatisticClient(user);
+            } catch (APIMgtUsageQueryServiceClientException e) {
+                throw new APIMgtUsageQueryServiceClientException("Error getting Statistics usage client instance", e);
+            }
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Use to check whether analytics is enabled
+     *
+     * @return return boolean value indicating whether analytics enable
+     */
+    public static boolean isDataPublishingEnabled() {
+        APIManagerAnalyticsConfiguration con = APIManagerAnalyticsConfiguration.getInstance();
+        return con.isAnalyticsEnabled();
+    }
+
+
+    /**
+     * Use to get instance of implementation class of the APIUsageStatisticsClient that is defined in the apim-manager.xml
+     *
+     * @return instance of a APIUsageStatisticsClient
+     * @throws APIMgtUsageQueryServiceClientException throws if instantiation problem occur
+     */
+    private static APIUsageStatisticsClient getStatisticClient(String user) throws APIMgtUsageQueryServiceClientException {
+
+        //read the api-manager.xml and get the Statistics class name
+        APIManagerConfiguration config = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService()
+                .getAPIManagerConfiguration();
+        String className = config.getFirstProperty("StatisticClientProvider");
+
+        try {
+
+            //get the Class from the class name
+            Class statClass=APIUtil.getClassFromName(className);
+            //use the constructor and pass appropriate args to get a instance
+            if(user!=null) {
+                usageStatisticsClient = (APIUsageStatisticsClient) statClass.getConstructor(String.class)
+                        .newInstance(user);
+            }else{
```

Comment:
fixed in commit: https://github.com/wso2/carbon-apimgt/commit/e34c4353fe49f6ce3ff615cb8ea5074d314f5e35